src/devices/OneWire/OneWireController.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Iot.Device.OneWire
+{
+    /// <summary>
+    /// Represents a 1-wire controller.
+    /// </summary>
+    public static partial class OneWireController
+    {
+        /// <summary>
+        /// Enumerate all 1-wire busses in the system.
+        /// </summary>
+        /// <returns>A list of discovered busses.</returns>
+        public static IEnumerable<OneWireBus> EnumerateBuses()
+        {
+            return EnumerateBusesInternal();
+        }
+
+        /// <summary>
+        /// Enumerate all devices found on 1-wire busses in this system.
+        /// </summary>
+        /// <param name="family">Family id used to filter devices.</param>
+        /// <returns>A list of devices found.</returns>
+        public static IEnumerable<OneWireDevice> EnumerateDevices(OneWireBus.DeviceFamily family = OneWireBus.DeviceFamily.Any)
+        {
+            foreach (var bus in EnumerateBuses())
+            {
+                foreach (var dev in bus.EnumerateDevices(family))
+                {
+                    yield return dev;
+                }
+            }
+        }
+
+        internal static OneWireDevice CreateDeviceByFamily(OneWireBus bus, string deviceId, OneWireBus.DeviceFamily family)
+        {
+            switch ((int)family)
```

Comment:
```csharp
switch (family)
{
    case OneWireBus.DeviceFamily.XYZ:
   ...
}
```

Reply:
There are no names except hex code for each family, but I will add each device I find as family name. You will get more device families back when specifying a device family with multiple devices in it.

Reply:
my comment is that enum is equivalent to int and there is no point in obscuring what the hex means by casting it to int

Reply:
My point was that it would no longer be a family id, but instead a pseudo device id and not possible to enumerate for example temperature devices. But I have changed all hex to enum values and added special out of spec values to simulate family ids.